### PR TITLE
Use bzip2 compressed feature set json as pipeline option

### PR DESCRIPTION
**Pull request by khorshuheng**
**What this PR does / why we need it**:
Dataflow runner has a limit of 256kb for pipeline option. As we are storing feature sets as json string in pipeline option, the size will grow proportionally to the number of feature set versions. Compressing the feature set json will help us to support more feature sets.

**Which issue(s) this PR fixes**:
None

**Does this PR introduce a user-facing change?**:
Users will be able to have more feature set before dataflow job submission fails. However, this depends on the compression ratio, which in turn depends on how much repetition exists in ithe feature set json.

### CODE REVIEW

**Comment by Yanson** [comment]
There are a lot of formatting changes in this. Which IDE + settings are you using?

I have imported the IntelliJ settings as described here:
https://github.com/gojek/feast/blob/master/docs/contributing.md#code-conventions

but I don't think it matches what you've submitted.

**Comment by khorshuheng** [comment]
> There are a lot of formatting changes in this. Which IDE + settings are you using?
> 
> I have imported the IntelliJ settings as described here:
> https://github.com/gojek/feast/blob/master/docs/contributing.md#code-conventions
> 
> but I don't think it matches what you've submitted.

It's the maven spotless plugin.

File: core/src/main/java/feast/core/util/ProtoUtil.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import com.google.protobuf.util.JsonFormat;
+import feast.core.FeatureSetProto;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProtoUtil {
+
+  public static String toJson(List<FeatureSetProto.FeatureSet> featureSets) throws IOException {
```
Comment (woop) [inline core/src/main/java/feast/core/util/ProtoUtil.java]: Can we avoid creating non-generic utility methods. `ProtoUtil` and `toJson` seem like a generic class and method, but the implementation is specific to FeatureSetProtos. 

Either we need to rename this to be more specific and generalize later, or move this functionality out.

File: ingestion/src/main/java/feast/ingestion/utils/CompressionUtil.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ */
+package feast.ingestion.utils;
+
+import java.io.*;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorOutputStream;
+
+public class CompressionUtil {
```
Comment (woop) [inline ingestion/src/main/java/feast/ingestion/utils/CompressionUtil.java]: Same as the comment above. Generic class name, generic method name, but the implementation isn't generic.

Comment (woop) [reply ingestion/src/main/java/feast/ingestion/utils/CompressionUtil.java]: Also don't see any comments.

File: core/src/main/java/feast/core/util/ProtoUtil.java (hunk 1 above)
Comment (khorshuheng) [reply core/src/main/java/feast/core/util/ProtoUtil.java]: Rename seems more of a band aid solution, so i refactored my commits such that it is no longer under util, and can be extended for other compression strategies.

**Pull request merged**